internal/core/assets.go

```diff
@@ -46,12 +46,16 @@ type AssetStatus struct {
 // a low-level logic function. Low-level functions explicitly take only the
 // AssetStatus to avoid accidental dependencies on non-logic attributes
 // like timestamps, UUIDs or error message strings.
-func StatusOfAsset(asset db.Asset) AssetStatus {
+func StatusOfAsset(asset db.Asset, cfg Config, res db.Resource) AssetStatus {
+	strictMaximumSize := asset.StrictMaximumSize
+	if val := cfg.MaxAssetSizeFor(res.AssetType, res.ScopeUUID); val != nil {
+		strictMaximumSize = val
+	}
```

Comment:
If both the asset and the MaxAssetSize thing dictate a limit, then the stronger limit should be used (i.e. the minimum of both).